chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1427 @@
+LIST_VOICE = ["Off", "English", "Chinese"]
+LIST_WORKMODE = ["Frequency", "Channel"]
+LIST_POWERON_DISPLAY_TYPE = ["LOGO", "BATT voltage"]
+LIST_BEEP = ["Off", "Beep", "Voice", "Both"]
+LIST_LANGUAGE = ["English", "Chinese"]
+LIST_SCANMODE = ["Time", "Carrier", "Search"]
+LIST_ALARMMODE = ["Local", "Send Tone", "Send Code"]
+LIST_MENU_QUIT_TIME = ["%s sec" % x for x in range(5, 55, 5)] + ["60 sec"]
+LIST_BACKLIGHT_TIMER = ["Always On"] + ["%s sec" % x for x in range(5, 25, 5)]
+LIST_ID_DELAY = ["%s ms" % x for x in range(100, 3100, 100)]
+LIST_QT_SAVEMODE = ["Both", "RX", "TX"]
+LIST_SKEY2_SHORT = ["FM", "Scan", "Search", "Vox"]
+LIST_RPT_TAIL_CLEAR = ["%s ms" % x for x in range(0, 1100, 100)]
+LIST_VOX_DELAY_TIME = ["%s ms" % x for x in range(500, 2100, 100)]
+LIST_VOX_LEVEL = ["Off"] + ["%s" % x for x in range(1, 10, 1)]
+LIST_VOX_LEVEL_ALT = ["%s" % x for x in range(1, 10, 1)]
+LIST_GPS_MODE = ["GPS", "Beidou", "GPS + Beidou"]
+LIST_GPS_TIMEZONE = ["%s" % x for x in range(-12, 13, 1)]
+
+TXP_CHOICES = ["High", "Low"]
+TXP_VALUES = [0x00, 0x02]
+
+STIMEOUT = 1.5
+
+
+def model_match(cls, data):
+    """Match the opened image to the correct version"""
+    return data[cls.MEM_TOTAL:] == bytes(cls.MODEL, 'utf-8')
+
+
+def _crypt(symbolIndex, buffer):
+    # Some weird encryption is used. From the table below, we only use "CO 7".
+    tblEncrySymbol = [b"BHT ", b"CO 7", b"A ES", b" EIY", b"M PQ",
+                      b"XN Y", b"RVB ", b" HQP", b"W RC", b"MS N",
+                      b" SAT", b"K DH", b"ZO R", b"C SL", b"6RB ",
+                      b" JCG", b"PN V", b"J PK", b"EK L", b"I LZ"]
+    tblEncrySymbols = tblEncrySymbol[symbolIndex]
+    decBuffer = b""
+    index1 = 0
+    for index2 in range(len(buffer)):
+        boolEncryptChar = ((tblEncrySymbols[index1] != 32) &
+                           (buffer[index2] != 0) &
+                           (buffer[index2] != 255) &
+                           (buffer[index2] != tblEncrySymbols[index1]) &
+                           (buffer[index2] != (tblEncrySymbols[index1] ^ 255))
+                           )
+        if (boolEncryptChar):
+            decByte = buffer[index2] ^ tblEncrySymbols[index1]
+            decBuffer += decByte.to_bytes(1, 'big')
+        else:
+            decBuffer += buffer[index2].to_bytes(1, 'big')
+        index1 = (index1 + 1) % 4
+    return decBuffer
+
+
+def _make_read_frame(addr, length):
+    """Pack the info in the header format"""
+    frame = _make_frame(b"\x52", addr, length)
+    # Return the data
+    return frame
+
+
+def _make_frame(cmd, addr, length, data=""):
+    """Pack the info in the header format"""
+    frame = cmd+struct.pack(">i", addr)[2:]+struct.pack("b", length)
+    # add the data if set
+    if len(data) != 0:
+        frame += data
+    # return the data
+    return frame
+
+
+def _do_ident(radio):
+    """Put the radio in PROGRAM mode & identify it"""
+    radio.pipe.baudrate = radio.BAUDRATE
+    radio.pipe.parity = "N"
```

Comment:
I do need to set the baudrate.. What is a better way to set it to 115200?

Reply:
It's set by the UI for you according to `radio.BAUDRATE` before it hands the serial device to the driver. So this should already be done for you, no need to re-do it (which resets the stack on Windows).

Reply:
Somehow that doesn't work... hmmm..

Reply:
It happens here: https://github.com/kk7ds/chirp/blob/master/chirp/wxui/clone.py#L135 .. Very few of the drivers manipulate the baudrate because it's already set for them. The exceptions are mostly drivers that probe for one of multiple allowed rates that the radio could be set to. So yeah, need to figure out what's going on there.

Reply:
Resolved. I used BAUDRATE which had to be BAUD_RATE